R/misc.R

```diff
@@ -617,17 +617,13 @@ check_nominal_type <- function(x, lvl) {
     was_factor <- fac_ref_cols[!(fac_ref_cols %in% fac_act_cols)]
 
     if (length(was_factor) > 0) {
-      rlang::warn(
-        paste0(
-          " There ",
-          ifelse(length(was_factor) > 1, "were ", "was "),
-          length(was_factor),
-          ifelse(length(was_factor) > 1, " columns ", " column "),
-          "that ",
-          ifelse(length(was_factor) > 1, "were factors ", "was a factor "),
-          "when the recipe was prepped:\n ",
-          paste0("'", was_factor, "'", collapse = ", "),
-          ".\n This may cause errors when processing new data."
+      cli::cli_warn(
+        c(
+          "!" = " There {?w/was/were} {length(was_factor)} column{?s} that \\
+                {?w/was a factor/were factors} when the recipe was prepped: \\
```

Comment:
```suggestion
                {?was a factor/were factors} when the recipe was prepped: \\
```

I believe we can drop this change both here and in the analogous part of the line above. The `?w` markup would only kick in when `length(was_factor) == 0`, which is never the case in here, but even in that case, just substituting in `"w"` is incorrect whereas the plural `"were factors"` reads fine.